### PR TITLE
Chore/add replica filter for query performance

### PR DESCRIPTION
**Pull request by joshenlim**
image

### CODE REVIEW

File: apps/studio/components/interfaces/QueryPerformance/QueryPerformance.tsx [hunk 1]
```diff
@@ -108,6 +111,10 @@ export const QueryPerformance = ({
     slowestExecutionTime,
   ])
 
+  useEffect(() => {
+    state.setSelectedDatabaseId(ref)
+  }, [])
```
Comment (alaister) [inline apps/studio/components/interfaces/QueryPerformance/QueryPerformance.tsx:116]: Are you intentionally omitting the `ref` from the deps? If so would you mind adding a comment on why 🙏🏻
```suggestion
  }, [ref])
```

File: apps/studio/hooks/analytics/useDbQuery.tsx [hunk 2]
```diff
@@ -38,12 +53,12 @@ const useDbQuery = (
     isRefetching,
     refetch,
   } = useQuery(
-    ['projects', project?.ref, 'db', { ...params, sql: resolvedSql }, where, orderBy],
+    ['projects', project?.ref, 'db', { ...params, sql: resolvedSql, identifier }, where, orderBy],
     ({ signal }) => {
       return executeSql(
         {
           projectRef: project?.ref,
```
Comment (alaister) [inline apps/studio/hooks/analytics/useDbQuery.tsx:60]: Just want to make sure this should still be `project?.ref` and not `identifier` still

File: apps/studio/components/interfaces/QueryPerformance/QueryPerformance.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/QueryPerformance/QueryPerformance.tsx:116]: good point actually - yeah can add ref in the deps

File: apps/studio/hooks/analytics/useDbQuery.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/hooks/analytics/useDbQuery.tsx:60]: yup so on line 45 if read replicas is enabled, it'll use the selectedDatabaseId, other default back to project.ref

in this case, each replica will have its own query performance results, hence why added the identifier param in the query key

**Review by alaister — COMMENTED** [review]
On a project with no read replicas I can still see the button and it seems to be constantly loading:
Screenshot 2024-05-02 at 15 15 22

**Comment by joshenlim** [comment]
ahh good catch sorry 🙏 pushed a fix! it shouldnt show up for projects that dont have replicas enabled for

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**